### PR TITLE
Turbopack: Use styled issue titles and details

### PR DESCRIPTION
**Pull request by wbinnssmith**
This uses styled issue titles and details introduced in vercel/turbo#6535, which also moves "Module not found" messaging to the title field for those issues.

Closes PACK-2013

### CODE REVIEW

**Review by kwonoj — DISMISSED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| buildDuration | 10.2s | 10.4s | ⚠️ +275ms |
| buildDurationCached | 6s | 6s | N/A |
| nodeModulesSize | 199 MB | 199 MB | ⚠️ +1.84 kB |
| nextStartRea..uration (ms) | 425ms | 438ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| 199-HASH.js gzip | 28.7 kB | 28.7 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 494.HASH.js gzip | 180 B | 181 B | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 240 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 100 kB | 100 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 501 B | 503 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | ✓ |
| edge-ssr-HASH.js gzip | 253 B | 255 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.28 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 311 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.17 kB | 3.17 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 483 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| index.html gzip | 527 B | 528 B | N/A |
| link.html gzip | 538 B | 542 B | N/A |
| withRouter.html gzip | 524 B | 523 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| edge-ssr.js gzip | 92.4 kB | 92.4 kB | N/A |
| page.js gzip | 145 kB | 145 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 626 B | 625 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 24.8 kB | 24.8 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/styled-title | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 167 kB | 167 kB | N/A |
| app-page-exp..prod.js gzip | 93.2 kB | 93.2 kB | N/A |
| app-page-tur..prod.js gzip | 94 kB | 94 kB | N/A |
| app-page-tur..prod.js gzip | 88.5 kB | 88.5 kB | N/A |
| app-page.run...dev.js gzip | 137 kB | 137 kB | N/A |
| app-page.run..prod.js gzip | 87.9 kB | 87.9 kB | N/A |
| app-route-ex...dev.js gzip | 23.8 kB | 23.8 kB | ✓ |
| app-route-ex..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route-tu..prod.js gzip | 16.5 kB | 16.5 kB | ✓ |
| app-route-tu..prod.js gzip | 16 kB | 16 kB | ✓ |
| app-route.ru...dev.js gzip | 23.2 kB | 23.2 kB | ✓ |
| app-route.ru..prod.js gzip | 16 kB | 16 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.8 kB | 21.8 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.8 kB | 21.8 kB | ✓ |
| server.runti..prod.js gzip | 49.1 kB | 49.1 kB | N/A |
| Overall change | 206 kB | 206 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: e027348687bf0c2443101b3e13ef6d0a8a9ef08f

**Comment by ijjk** [comment]
## Failing test suites 
Commit: beb5164c2cd2915c2b742c57c2403c224279c372

`pnpm test-dev test/development/acceptance/ReactRefreshRequire.test.ts` 
- ReactRefreshRequire > propagates a module that stops accepting in next version

<details>
<summary>Expand output</summary>

  ● ReactRefreshRequire › propagates a module that stops accepting in next version

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

      Array [
    -   "init BarV2",
    +   "init FooV1",
        "init BarV2",
        "init FooV1",
      ]

      423 |     // it didn't export a component, so we go higher.
      424 |     // We stop at Foo which currently _does_ export a component.
    > 425 |     expect(await session.evaluate(() => (window as any).log)).toEqual([
          |                                                               ^
      426 |       // Bar evaluates twice:
      427 |       // 1. To invalidate itself once it realizes it's no longer acceptable.
      428 |       // 2. As a child of Foo re-evaluating.

      at Object.toEqual (development/acceptance/ReactRefreshRequire.test.ts:425:63)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by kwonoj — DISMISSED** [review]

File: test/development/basic/next-rs-api.test.ts [hunk 1]
```diff
@@ -23,11 +24,38 @@ function normalizePath(path: string) {
...
+    case 'strong':
+      return '**' + styled.value + '**'
+    case 'code':
+      return '`' + styled.value + '`'
+    case 'line': {
+      let line = ''
+      for (const part of styled.value) {
+        line += styledStringToMarkdown(part)
+      }
+      return line + '\n'
```
Comment (sokra) [inline test/development/basic/next-rs-api.test.ts:40]:
```suggestion
      return line
```

A line should not add a line break. That's the job of `stack`

File: test/development/basic/next-rs-api.test.ts [hunk 2]
```diff
@@ -23,11 +24,38 @@ function normalizePath(path: string) {
...
+    }
+    case 'stack':
+      let stack = ''
+      for (const part of styled.value) {
+        stack += styledStringToMarkdown(part) + '\n'
+      }
+      return stack + '\n'
```
Comment (sokra) [inline test/development/basic/next-rs-api.test.ts:47]:
```suggestion
      return styled.value.map(styledStringToMarkdown).join('\n')
```

No tailing new line

File: packages/next-swc/crates/napi/src/next_api/utils.rs [hunk 3]
```diff
@@ -100,9 +100,9 @@ pub struct NapiIssue {
     pub severity: String,
     pub category: String,
     pub file_path: String,
-    pub title: String,
+    pub title: serde_json::Value,
```
Comment (sokra) [inline packages/next-swc/crates/napi/src/next_api/utils.rs:103]:
```suggestion
    pub title: NapiStyledString,
```

File: packages/next-swc/crates/napi/src/next_api/utils.rs [hunk 4]
```diff
@@ -100,9 +100,9 @@ pub struct NapiIssue {
...
     pub description: serde_json::Value,
```
Comment (sokra) [inline packages/next-swc/crates/napi/src/next_api/utils.rs:104]:
```suggestion
    pub description: Option<NapiStyledString>,
```

File: packages/next-swc/crates/napi/src/next_api/utils.rs [hunk 5]
```diff
@@ -100,9 +100,9 @@ pub struct NapiIssue {
...
-    pub detail: String,
+    pub detail: serde_json::Value,
```
Comment (sokra) [inline packages/next-swc/crates/napi/src/next_api/utils.rs:105]:
```suggestion
    pub detail: Option<NapiStyledString>,
```

File: packages/next/src/build/swc/index.ts [hunk 6]
```diff
@@ -499,9 +499,9 @@ export interface Issue {
   severity: string
   category: string
   filePath: string
-  title: string
-  description: StyledString
-  detail: string
+  title: StyledString
+  description: StyledString | null
+  detail: StyledString | null
```
Comment (sokra) [inline packages/next/src/build/swc/index.ts:504]:
```suggestion
  title: StyledString
  description?: StyledString
  detail?: StyledString
```

File: test/development/basic/next-rs-api.test.ts (hunk 2 above)
Comment (sokra) [reply test/development/basic/next-rs-api.test.ts:47]: Similar changes to `renderStyledStringToErrorAnsi`

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]

**Pull request merged**